public/api.js

```diff
@@ -0,0 +1,113 @@
+export { getCategories, getToken, getPlaylists, search }
+
+/**
+ * Функция для получения токена доступа 
+ * При возникновении ошибки в консоль выводится сообщение
+ * @returns токен доступа 
+*/
+async function getToken() {
+  const client_id = 'cc54ff5069b04ecbb4cf473f445e9872';
+  const client_secret = '5a84d500c04542fe87ca70dee98313fc';
+  try {
+    const response = await fetch('https://accounts.spotify.com/api/token', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'Authorization': 'Basic ' + btoa(client_id + ':' + client_secret)
+      },
+      body: 'grant_type=client_credentials'
+    })
+    .then((res) => {
```

Comment:
Можно же получить результат запроса, а после получить данные
```suggestion
   if (!responce.ok){
      ...
   } else {
      const json = await responce.json();
      return json.access_token;
   }
```